pkg/plugins/discovery_unix.go

```diff
@@ -2,5 +2,21 @@
 // +build !windows
 
 package plugins // import "github.com/docker/docker/pkg/plugins"
+import (
+	"github.com/docker/docker/pkg/homedir"
+	"github.com/docker/docker/rootless"
+	"path/filepath"
+)
 
 var specsPaths = []string{"/etc/docker/plugins", "/usr/lib/docker/plugins"}
+
+// ConfigureSpecsPaths configures plugins.specsPaths depending on if the daemon
+// is running in rootless mode or not. On Windows, it is a no-op.
+func ConfigureSpecsPaths() {
+	if rootless.RunningWithRootlessKit() {
+		dataHome, err := homedir.GetDataHome()
+		if err == nil {
+			specsPaths = []string{filepath.Join(dataHome, "docker/plugins")}
```

Comment:
Personally I would prefer to avoid functions that mutate global variables and depend on being called in a specific time and place.
I think it would be better to change `specsPaths` into a function returning paths, which would be defined differently in each `discovery_<platform>.go`. This way we don't have any global mutable variable and don't need to worry about calling the `ConfigureSpecsPaths`.
What do you think?

Reply:
Sounds good, will do

Reply:
Ah, yes, was thinking along the same lines as well; was also wondering; would it do any harm to unconditionally append `homedir.GetDataHome()` to the list of paths to look in? Perhaps needs a `sync.Once` to resolve the path once (if there's nothing like that yet), but then a `specPaths()` function could return the lists of paths to look for.

Reply:
^^ I didn't dig into all code-paths though, so be sure to look if that idea works 😅

Reply:
I introduced `func SpecsPaths() []string` while keeping `specsPaths` around to avoid a large diff in the Unit-Tests.

> would it do any harm to unconditionally append `homedir.GetDataHome()` to the list of paths to look in? 

Right now the user can use his own plugins without being affected by plugins installed in the host system.
If we add `$XDG_DATA_HOME/.local/share/docker/plugins` to `SpecsPaths` unconditionally, the user no longer has this option